src/query/expression/src/types/geometry.rs

```diff
@@ -0,0 +1,169 @@
+// Copyright 2021 Datafuse Labs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::ops::Range;
+
+use super::binary::BinaryColumn;
+use super::binary::BinaryColumnBuilder;
+use super::binary::BinaryIterator;
+use crate::property::Domain;
+use crate::types::ArgType;
+use crate::types::DataType;
+use crate::types::GenericMap;
+use crate::types::ValueType;
+use crate::values::Column;
+use crate::values::Scalar;
+use crate::values::ScalarRef;
+use crate::ColumnBuilder;
+
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct GeometryType;
+
+impl ValueType for GeometryType {
+    type Scalar = Vec<u8>;
+    type ScalarRef<'a> = &'a [u8];
+    type Column = BinaryColumn;
```

Comment:
Maybe we can have `Vec<Geometry>` as column represention instead of binary.

Reply:
Maybe in next pr #14553